File: JAVA/TowerOfHanoi.java
public class TowerOfHanoi {
    public static void main(String[] args) {
        int numberOfDisks = 3; // Change this to the desired number of disks.
        towerOfHanoi(numberOfDisks, 'A', 'B', 'C');
    }

    public static void towerOfHanoi(int n, char sourceRod, char auxiliaryRod, char targetRod) {
        if (n == 1) {
            // If there's only one disk, move it from the source rod to the target rod.
            System.out.println("Move disk 1 from " + sourceRod + " to " + targetRod);
            return;
        }

        // Move n-1 disks from the source rod to the auxiliary rod using the target rod as the auxiliary.
        towerOfHanoi(n - 1, sourceRod, targetRod, auxiliaryRod);

        // Move the nth disk from the source rod to the target rod.
        System.out.println("Move disk " + n + " from " + sourceRod + " to " + targetRod);

        // Move the n-1 disks from the auxiliary rod to the target rod using the source rod as the auxiliary.
        towerOfHanoi(n - 1, auxiliaryRod, sourceRod, targetRod);
    }
}

File: c++/tower_of_hanoi.cpp
#include <iostream>
using namespace std;
void TowerOfHanoi(int n, char source, char aux, char dest) {
    if (n == 1) {
        cout << "Move disk 1 from " << source << " to " << dest << endl;
        return;
    }
    TowerOfHanoi(n - 1, source, dest, aux);
    cout << "Move disk " << n << " from " << source << " to " << dest << endl;
    TowerOfHanoi(n - 1, aux, source, dest);
}
int main() {
    int NumDisks = 3;
    char source = 'A';
    char aux = 'B';
    char dest = 'C';
    cout << "Tower of Hanoi solution with " << NumDisks << " disks:" << endl;
    TowerOfHanoi(NumDisks, source, aux, dest);
    return 0;
}
